smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs

```diff
@@ -217,58 +259,18 @@ pub fn do_swap_deposit_merge(
     target_lower_tick: i64,
     target_upper_tick: i64,
     tokens_provided: &[Coin],
-    position_id: Option<u64>,
     ratio_of_swappable_funds_to_use: Decimal,
     twap_window_seconds: u64,
 ) -> Result<Response, ContractError> {
-    if SWAP_DEPOSIT_MERGE_STATE.may_load(deps.storage)?.is_some() {
-        return Err(ContractError::SwapInProgress {});
-    }
-
     let swap_amounts: Result<Vec<_>, _> = tokens_provided
         .iter()
         .map(|c| c.amount.checked_mul_floor(ratio_of_swappable_funds_to_use))
         .collect();
     let swap_amounts = swap_amounts?;
 
-    let mut target_range_position_ids = vec![];
-    if let Some(pos_id) = position_id {
-        target_range_position_ids.push(pos_id);
-    }
-
-    SWAP_DEPOSIT_MERGE_STATE.save(
-        deps.storage,
-        &SwapDepositMergeState {
-            target_lower_tick,
-            target_upper_tick,
-            target_range_position_ids,
-        },
-    )?;
-
     let pool_config = POOL_CONFIG.load(deps.storage)?;
     let pool_details = get_cl_pool_info(&deps.querier, pool_config.pool_id)?;
 
-    let response = Response::new()
-        .add_attribute("method", "reply")
-        .add_attribute("action", "do_swap_deposit_merge");
-    if swap_amounts.iter().all(|amount| amount.is_zero()) {
-        let position = POSITION.load(deps.storage)?;
-
-        // if we have not tokens to swap, that means all tokens we correctly used in the create position
-        // this means we can save the position id of the first create_position
-        POSITION.save(
-            deps.storage,
-            &Position {
-                position_id: position_id.expect("position id should be set if no swap is needed"),
-                join_time: position.join_time,
-                claim_after: position.claim_after,
-            },
-        )?;
-
-        SWAP_DEPOSIT_MERGE_STATE.remove(deps.storage);
```

Comment:
We should still make sure to clean up our state item after we're done using it

Reply:
The cleanup happens in `handle_iteration_create_position_reply`. Here it was just deleted as it was written before without needing it.

Reply:
Sounds good, misread then